src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -1534,12 +1535,28 @@ def aks_upgrade(cmd,    # pylint: disable=unused-argument
                 client,
                 resource_group_name,
                 name,
-                kubernetes_version,
+                kubernetes_version='',
                 control_plane_only=False,
                 no_wait=False,
+                node_image_only=False,
                 **kwargs):  # pylint: disable=unused-argument
+    from knack.prompting import prompt_y_n
     instance = client.get(resource_group_name, name)
 
+    if kubernetes_version != '' and node_image_only:
+        raise CLIError('Conflicting flags. Upgrading the Kubernetes version will also upgrade node image version. If you only want to upgrade the node version please use the "--node-image-only" option only.')
+
+    if node_image_only:
+        msg = "This node image upgrade operation will run across every node pool in the cluster and might take a while, do you wish to continue?"
+        if not prompt_y_n(msg, default="n"):
+            return None
+        agent_pool_client = cf_agent_pools(cmd.cli_ctx)
+        for agent_pool_profile in instance.agent_pool_profiles:
+            if agent_pool_profile.type.lower() == "availabilityset":
```

Comment:
This is done down on 1572. Just move it up and reuse vmas_cluster?

Reply:
sure